anta/tests/bfd.py

```diff
@@ -285,3 +285,83 @@ def test(self) -> None:
         if up_failures:
             up_failures_str = "\n".join(up_failures)
             self.result.is_failure(f"\nFollowing BFD peers were down:\n{up_failures_str}")
+
+
+class VerifyBFDPeersRegProtocols(AntaTest):
+    """Verifies BFD peer(s) has the specified protocol(s) registered.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if IPv4 BFD peers are configured with specified protocol(s).
+    * Failure: The test will fail if IPv4 BFD peers are not found, BFD session is not established, specified protocol(s) are not registered on the BFD peer.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.bfd:
+      - VerifyBFDPeersRegProtocols:
+          bfd_peers:
+            - peer_address: 192.0.255.7
+              vrf: default
+              protocols:
+                - bgp
+    ```
+    """
+
+    name = "VerifyBFDPeersRegProtocols"
+    description = "Verifies IPv4 BFD peer(s) has the specified protocol(s) registered."
+    categories: ClassVar[list[str]] = ["bfd"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show bfd peers detail", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyBFDPeersRegProtocols test."""
+
+        bfd_peers: list[BFDPeer]
+        """List of IPv4 BFD peers."""
+
+        class BFDPeer(BaseModel):
+            """Model for an IPv4 BFD peer."""
+
+            peer_address: IPv4Address
+            """IPv4 address of a BFD peer."""
+            vrf: str = "default"
+            """Optional VRF for BFD peer. If not provided, it defaults to `default`."""
+            protocols: list[Literal["bgp", "isis", "ospf", "eigrp"]]
```

Comment:
There are more protocols available. Please confirm:
```
DC1-LEAF1A#show bfd peers protocol ?
  bgp           
  isis          
  lag           
  ospf          
  ospfv3        
  pim           
  route-input   
  static-bfd    
  static-route  
  vrrp          
  vxlan
  ```

Reply:
Added these protocol under BfdProtocol in custom_types.